Modules/_json.c

```diff
@@ -1518,8 +1518,14 @@ encoder_listencode_dict(PyEncoderObject *s, _PyUnicodeWriter *writer,
     PyObject *key, *value;
     bool first = true;
 
-    if (PyDict_GET_SIZE(dct) == 0)  /* Fast path */
-        return _PyUnicodeWriter_WriteASCIIString(writer, "{}", 2);
+    if (PyDict_CheckExact(dct)) {
+        if (PyDict_GET_SIZE(dct) == 0)  /* Fast path */
+            return _PyUnicodeWriter_WriteASCIIString(writer, "{}", 2);
```

Comment:
```suggestion
        if (PyDict_GET_SIZE(dct) == 0) {  /* Fast path */
            return _PyUnicodeWriter_WriteASCIIString(writer, "{}", 2);
        }
```
See https://peps.python.org/pep-0007/#code-lay-out